### PR TITLE
Move info on contributing to docs out of toplevel readme

### PR DESCRIPTION
**Pull request by ozdanborne**
I remove the part on "finding docs-archive". Figured enough time has passed that we can get rid of that.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,149 +14,3 @@ Note that the README in this repo is targeted at Calico docs contributors.
 
 **If you are looking for the repository formerly known as `projectcalico/calico`,
 it has been renamed to [`projectcalico/felix`](https://github.com/projectcalico/felix).**
-
```
Comment (caseydavenport) [inline README.md]: I think you should update the title of this page from "Project Calico Documentation" to simply "Calico"

File: README.md [hunk 2]
```diff
@@ -14,149 +14,3 @@ Note that the README in this repo is targeted at Calico docs contributors.
 
 **If you are looking for the repository formerly known as `projectcalico/calico`,
```
Comment (caseydavenport) [inline README.md]: I think we can just remove this line at this point - WDYT?

File: README.md (hunk 1 above)
Comment (caseydavenport) [inline README.md]: Also, I think this part should be updated:

```
 <blockquote>
 Note that the README in this repo is targeted at Calico docs contributors.
 <h1>Documentation for Calico users is here:<br><a href="http://docs.projectcalico.org">http://docs.projectcalico.org</a></h1>
 </blockquote>
```

To instead say something like:

```
See [contributing to the Calico docs](...) for information on how to make changes to the docs.projectcalico.org website.
```

**Comment by ozdanborne** [comment]
ping @caseydavenport , I rebased this and applied your suggestions.

**Pull request merged**